archean/egg-archean.json

```diff
@@ -0,0 +1,174 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PANEL",
+    "meta": {
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2024-10-21T00:55:52+00:00",
+    "name": "Archean",
+    "author": "tueem@tomatentum.net",
+    "uuid": "01555bd3-155f-4ab6-b63d-c8bed0a3585f",
+    "description": "Archean invites you to unleash your creativity alone or with friends in a creative sandbox mode.Design and build vehicles, bases, rockets, space stations and much more using customizable modular blocks.Create unique configurations using components, program their behavior and explore an environment free of constraints.\\nTest your builds, refine your designs and defy the laws of physics.Archean is a game in development focused on engineering and build.",
+    "features": [],
+    "docker_images": {
+        "Ubuntu": "ghcr.io\/pelican-eggs\/yolks:ubuntu"
+    },
+    "file_denylist": [],
+    "startup": ".\/archean_start.sh",
```

Comment:
Startup commands that call a script are not typically allowed in this repo, because a rogue user can change it to make the container do whatever they want. Everything you do in `archean_start.sh` can easily be done via the Egg's Installation Script or (for config values) [Configuration Files section](https://pterodactyl.io/community/config/eggs/creating_a_custom_egg.html#configuration-files) with the INI parser. It would be no less user friendly. If anything, it would be more user friendly to admins who could glance at the Egg and understand what it is doing vs. being clueless as to what this remotely downloaded script entails (which could also be maliciously changed at any time). Therefore, you need to move all functionality to be within the Egg or this repo. This means you also need to move your sample `server.ini` to be included with this PR and be downloaded from the future permalink it would reside at within this repo (if it were to be merged into `main`).

Reply:
But what would be worse in my opinion is that the variables "$GAME_MODE" would be set by "0" or 1" and "$ONLINE_MODE" can't be Boolean because it needs to be either "yes or "no".
Or is there some feature I don't know of to map these values?

Reply:
In my opinion that would make it a worse user experience because yes/no wouldn't be turned into a button and 0 and 1 are also bad gamemode descriptors. But if it is required to be merged then I will change it.

Reply:
I get that but in my opinion yes/no should not have been used by the game devs but 0 & 1 or true and false

Reply:
Ah I just tested 0/1 works instead of yes/no. That's not mentioned in their docs

Reply:
> yes/no wouldn't be turned into a switch

Correct, but it *would* be turned into a dropdown box select, which is still pretty good in my opinion. However, if the variable supports `boolean` that should be fine; hopefully the devs don't enforce yes/no in the future.

> how do i replace the whole contents of a file with a egg variable?

Unless the file is single-line (ie. comma-separated), there's unfortunately no way to replace a multi-line file with an Egg's Configuration Files. How you're doing it in the startup command now is fine though.